### PR TITLE
转换规则No66-70 torch.nn.functional.pairwise_distance

### PR DESCRIPTION
**Pull request by Liyulingyue**
### PR Docs

None

### PR APIs

```bash
torch.nn.functional.cosine_embedding_loss
torch.nn.functional.gumbel_softmax
torch.nn.functional.hinge_embedding_loss
torch.nn.functional.pairwise_distance
torch.nn.functional.poisson_nll_loss
```

### API Discriptions
https://github.com/PaddlePaddle/PaConvert/issues/112 case 67
poisson_nll_loss 参数 full会导致计算差异

### CODE REVIEW

File: paconvert/api_mapping.json [hunk 1]
```diff
@@ -8732,5 +8732,103 @@
...
+      "reduce",
+      "reduction"
+    ],
+    "kwargs_change": {
+      "target": "label",
+      "tau": "temperature",
+      "dim": "axis",
+      "eps": "epsilon"
+    },
+    "unsupport_args": [
```
Comment (Vooolta) [inline paconvert/api_mapping.json:8790]: 参数可以绕过，可以参考SizeAverageMatcher用process_reduce_and_size_average(kwargs)实现一下

File: paconvert/api_mapping.json [hunk 2]
```diff
@@ -8732,5 +8732,103 @@
...
+      "target",
+      "margin",
+      "size_average",
+      "reduce",
+      "reduction"
+    ],
+    "kwargs_change": {
+      "target": "label"
+    },
+    "unsupport_args": [
```
Comment (Vooolta) [inline paconvert/api_mapping.json:8810]: 参数可以绕过，可以参考SizeAverageMatcher用process_reduce_and_size_average(kwargs)实现一下

File: paconvert/api_mapping.json [hunk 3]
```diff
@@ -8732,5 +8732,103 @@
...
```
Comment (Vooolta) [inline paconvert/api_mapping.json:8829]: 参数可以绕过，可以参考SizeAverageMatcher用process_reduce_and_size_average(kwargs)实现一下

File: tests/test_nn_functional_gumbel_softmax.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        import torch
+        a = [[1.3192, 1.9915, 1.9674, 1.7151],[1.3492, 0.1915, 2.9434, 1.4151]]
+        x = torch.tensor(a)
+        result = torch.nn.functional.gumbel_softmax(x, tau=2, hard=True)
+        """
+    )
+    obj.run(pytorch_code, ["result"], check_value=False)
+
+
+def test_case_4():
```
Comment (zhwesky2010) [inline tests/test_nn_functional_gumbel_softmax.py:59]: 可以加个eps的case，使用unsupport=True来测试

File: tests/test_nn_functional_gumbel_softmax.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+def test_case_1():
+    pytorch_code = textwrap.dedent(
+        """
+        import torch
+        a = [[1.3192, 1.9915, 1.9674, 1.7151],[1.3492, 0.1915, 2.9434, 1.4151]]
+        x = torch.tensor(a)
+        result = torch.nn.functional.gumbel_softmax(x)
+        """
+    )
+    obj.run(pytorch_code, ["result"], check_value=False)
```
Comment (zhwesky2010) [inline tests/test_nn_functional_gumbel_softmax.py:32]: 这个为什么计算的值不同呢

File: tests/test_nn_functional_poisson_nll_loss.py [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+        b = [[1.3492, 0.1915, 2.9434, 1.4151]]
+        x1 = torch.tensor(a)
+        x2 = torch.tensor(b)
+        result = torch.nn.functional.poisson_nll_loss(x1, x2, log_input=False)
+        """
+    )
+    obj.run(pytorch_code, ["x1", "x2", "result"])
+
+
+def _test_case_3():
```
Comment (zhwesky2010) [inline tests/test_nn_functional_poisson_nll_loss.py:51]: paddle的bug记得在任务那个页面写出来，我们会定期收集，并集中修复

Comment (zhwesky2010) [reply tests/test_nn_functional_poisson_nll_loss.py:51]: 这个是什么原因禁用呢，需要在上面注释写明原因

**Comment by luotao1** [comment]
需要解决下CodeStyle流水线问题

**Comment by zhwesky2010** [comment]
@Liyulingyue 上面的评论回复一下

File: tests/test_nn_functional_gumbel_softmax.py (hunk 5 above)
Comment (Liyulingyue) [reply tests/test_nn_functional_gumbel_softmax.py:32]: https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/nn/functional/gumbel_softmax_cn.html#gumbel-softmax
根据对gumbel的介绍，这个计算流程是具有一定随机性的

File: tests/test_nn_functional_poisson_nll_loss.py (hunk 6 above)
Comment (Liyulingyue) [reply tests/test_nn_functional_poisson_nll_loss.py:51]: 不知道啥情况，计算结果torch是
tensor(3.8456)，paddle是
Tensor(shape=[], dtype=float32, place=Place(cpu), stop_gradient=True,
       3.77250624)
按照计算公式来说应该一样的

File: paconvert/api_mapping.json [hunk 7]
```diff
@@ -7701,6 +7767,26 @@
...
+      "input",
+      "target",
+      "log_input",
+      "full",
+      "size_average",
+      "eps",
+      "reduce",
+      "reduction"
+    ],
+    "kwargs_change": {
```
Comment (zhwesky2010) [inline paconvert/api_mapping.json:7783]: 这个是不是写错了，torch的没有dim、tau，paddle没有epsilon、axis、temperature

https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/nn/functional/poisson_nll_loss_cn.html#poisson-nll-loss

File: tests/test_nn_functional_poisson_nll_loss.py (hunk 6 above)
Comment (zhwesky2010) [reply tests/test_nn_functional_poisson_nll_loss.py:52]: 这个错误可以再确认一下，是不是full=True时存在计算bug，我们会记录相应bug

Comment (zhwesky2010) [reply tests/test_nn_functional_poisson_nll_loss.py:52]: > 这个错误可以再确认一下，是不是full=True时存在计算bug，我们会记录相应bug

这个错误可以再确认一下，是不是full=True时存在计算bug，我们会记录相应bug

**Comment by zhwesky2010** [comment]
这个错误可以再确认一下，是不是full=True时存在计算bug，我们会记录相应bug

**Comment by luotao1** [comment]
@Liyulingyue 请补充下转换文档

**Comment by luotao1** [comment]
@Liyulingyue 请解决下冲突和回答下上面的问题

File: paconvert/api_mapping.json (hunk 7 above)
Comment (Liyulingyue) [reply paconvert/api_mapping.json:7783]: 是的，我改一下~

File: paconvert/api_mapping.json [hunk 8]
```diff
@@ -7701,6 +7767,26 @@
...
+      "target": "label",
+      "eps": "epsilon",
+      "dim": "axis",
+      "tau": "temperature"
```
Comment (Liyulingyue) [inline paconvert/api_mapping.json:7787]:
```suggestion
      "target": "label",
      "eps": "epsilon"
```

File: tests/test_nn_functional_gumbel_softmax.py (hunk 4 above)
Comment (Liyulingyue) [reply tests/test_nn_functional_gumbel_softmax.py:59]: added~

**Comment by Liyulingyue** [comment]
> Member

solved~

**Comment by zhwesky2010** [comment]
映射文档写了吗？

**Comment by Liyulingyue** [comment]
> 映射文档写了吗？

应该还没有，我稍后确认一下~

**Comment by zhwesky2010** [comment]
> > 映射文档写了吗？
> 
> 应该还没有，我稍后确认一下~

这个需要先写映射关系文档，再写转换规则的
infoflow 2023-08-23 21-26-37

**Comment by zhwesky2010** [comment]
@Liyulingyue 你好，请补充一下映射文档哈，相当于转写思路的描述。不然会造成文档缺失，这个后面如果再让其他同学来写，肯定也没有你对这几个API的转写思路更清楚~

**Comment by luotao1** [comment]
> torch.nn.functional.cosine_embedding_loss 文档已存在
torch.nn.functional.gumbel_softmax https://github.com/PaddlePaddle/docs/pull/6146
torch.nn.functional.hinge_embedding_loss 文档已存在
torch.nn.functional.pairwise_distance 文档已存在
torch.nn.functional.poisson_nll_loss 文档已存在

@Liyulingyue 每一行标一下No.xxx吧

File: paconvert/api_mapping.json (hunk 7 above)
Comment (zhwesky2010) [reply paconvert/api_mapping.json:8875]: @Liyulingyue 有些文档都有bug，需要仔细核对下，如果文档有bug需要进行修复

infoflow 2023-08-29 11-09-46


infoflow 2023-08-29 11-10-56

**Comment by zhwesky2010** [comment]
> 

请问 No. 67 torch.nn.functional.pairwise_distance、No. 68 torch.nn.functional.poisson_nll_loss 的文档在哪里

**Comment by zhwesky2010** [comment]
> > torch.nn.functional.cosine_embedding_loss 文档已存在
> > torch.nn.functional.gumbel_softmax [PaddlePaddle/docs#6146](https://github.com/PaddlePaddle/docs/pull/6146)
> > torch.nn.functional.hinge_embedding_loss 文档已存在
> > torch.nn.functional.pairwise_distance 文档已存在
> > torch.nn.functional.poisson_nll_loss 文档已存在
> 
> @Liyulingyue 每一行标一下No.xxx吧

请问 No. 67 torch.nn.functional.pairwise_distance、No. 68 torch.nn.functional.poisson_nll_loss 的文档在哪里

**Comment by zhwesky2010** [comment]
@Liyulingyue 这个PR还有些问题，先合入了我来修吧

**Review by zhwesky2010 — APPROVED** [review]
